docs/merger_architecture.md

```diff
@@ -1,172 +0,0 @@
-# Merger architecture
-
-The merger is a new application that we are planning to write in Q3 2018.
-
-It receives source records/transformed Works, and it works out which source records correspond to the same Work in our pipeline.
-
-This document has some notes on the proposed architecture.
-
-## Model
-
-We model source records and their relationships as a graph.
-The vertices are the source records, and there is a directed edge from X to Y if information in the source record X tells us that it is related to Y.
-
-Each merged work is one of the connected components of this graph.
-
-![](relater_graph.png)
-
-In this example, there are three works:
-
--   A red work made of three source records
--   A green work made of two source records
--   A blue work that is a single source record
-
-The relater will receive updates to this graph, one vertex at a time.
-When it receives an update, it needs to tell us:
-
--   Any new works which have been created
--   Any old works which have been destroyed and need to be redirected (if two components emerge, or one component splits)
-
-## Storage
-
-We considered three storage options:
-
--   Neo4j (a filesystem-backed graph database)
--   Titan/JanusGraph (a DynamoDB-backed graph database)
--   Storing graph information in SQL
-
-We decided to use SQL, specifically RDS, as we already use it in the ID minter, and we know how it scales.
-
-The idea is to store the structure of the graph in SQL, but do all of the graph theory logic outside the database.
-Trying to do graph operations in SQL queries feels a step too far!
-
-## Database schema
-
-We can identify subgraphs/components by concatenating the sorted IDs of the vertices.
-For example, the graph below has three components: `A-B-C-D`, `E-F-G` and `H`.
```

Comment:
Is the fact that sorted vertex IDs are used to identify a graph used?  Otherwise perhaps a unique graph id could be easier.

Reply:
I prefer the deterministic method here. Generating unique ids and keeping track of them is I think a harder problem.